File: Array_and_Hashing/JS/PascalTraingle.js
init();


function pascalTriangle(n) {
    let temp = [];
    let result = [];

    for (let i = 0; i < n; i++) {
        for (let j = 0; j <= i; j++) {
            if (i === j) {
                temp.push(1);
                continue;
            } else if (j === 0) {
                temp.push(1);
                continue;
            }
            temp.push(result[i - 1][j - 1] + result[i - 1][j]);
        }
        result.push(temp.slice());
        temp = [];
    }
    return result;
}

function init() {
    let n = 5;
    let result = pascalTriangle(n);
    console.log(...result)
}

File: Array_and_Hashing/PYTHON/PascalTraingle.py
# 1
# 1 1
# 1 2 1
# 1 3 3 1
# 1 4 6 4 1

def pascalTriangle(level):
    res = []

    for i in range(0,level):
        temp = []
        for j in range(0,i+1): # strange Behaviour of loops in python
            if i == j:
                temp.append(1)
            elif j == 0:
                temp.append(1)
            else :
                temp.append(res[i-1][j-1] + res[i-1][j])
        res.append(temp)        
    return res


print(pascalTriangle(5))                    

